mscore/palette/palettelistview.cpp

```diff
@@ -0,0 +1,108 @@
+//=============================================================================
+//  PaletteListView
+//
+//  Copyright (C) 2020 Peter Jonas
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2
+//  as published by the Free Software Foundation and appearing in
+//  the file LICENCE.GPL
+//=============================================================================
+
+#include "palettelistview.h"
+
+#include "palettemodel.h"
+
+namespace Ms {
+
+//---------------------------------------------------------
+//   PaletteListView::PaletteListView
+//---------------------------------------------------------
+
+PaletteListView::PaletteListView(PalettePanel* panel, QWidget* parent)
+      : QListView(parent)
+      {
+      setViewMode(QListView::IconMode);
+      setMovement(QListView::Static);
+      setResizeMode(QListView::Adjust);
+      setIconSize(panel->gridSize());
+      setSpacing(-5); // zero spacing still leaves a large gap between icons
+
+      PaletteTree* tree = new PaletteTree();
+      tree->append(panel);
+
+      PaletteTreeModel* model = new PaletteTreeModel(tree);
+      QModelIndex parentCategory = model->index(0, 0, QModelIndex());
+
+      setModel(model);
+      setRootIndex(parentCategory);
+      }
+
+//---------------------------------------------------------
+//   PaletteListView::currentCell
+//---------------------------------------------------------
+
+const PaletteCell* PaletteListView::currentCell() const
+      {
+      return model()->data(currentIndex(), PaletteTreeModel::PaletteCellRole).value<const PaletteCell*>();
+      }
+
+//---------------------------------------------------------
+//   PaletteListView::currentElement
+//---------------------------------------------------------
+
+Element* PaletteListView::currentElement() const
+      {
+      return currentCell()->element.get();
+      }
+
+//---------------------------------------------------------
+//   PaletteListView::focusNextMatchingCell
+//---------------------------------------------------------
+
+void PaletteListView::focusNextMatchingCell(const QString& str)
+      {
+      const int nextRow = (currentRow() == count() - 1) ? 0 : currentRow() + 1;
+      const QModelIndex nextIndex = model()->index(nextRow, 0, rootIndex());
+      const auto matchedIndexList = model()->match(nextIndex, Qt::ToolTipRole, str);
+      if (!matchedIndexList.isEmpty())
+            setCurrentIndex(matchedIndexList.first());
+      }
+
+//---------------------------------------------------------
+//   onlyContainsVisibleCharacters
+// Return true iff string is non-empty and
```

Comment:
Could you please make these comments (except for the function name itself) start from three slashes (`///`) to make them visible by Doxygen? This isn't an "official" convention right now but still should be useful, I believe.

Also there is a typo in `if` in this line.

Reply:
@dmitrio95, sure. BTW, `iff` is not a typo. It is short for ["if and only if"](https://en.wikipedia.org/wiki/If_and_only_if).

Reply:
better spell that out, it is not a widely known acronym

Reply:
@dmitrio95, done!